main.py

```diff
@@ -1,65 +1,340 @@
+headers = list(fieldmap.keys())
 
-def create_date_time (row):
-    date_time = row['issue_date'] + ' ' + row['issue_time']
-    return datetime.strptime(date_time, '%m/%d/%Y %H:%M').strftime('%Y-%m-%d %H:%M:%S')
+def parse_lat_lon(value):
+    value = float(value)
+    if value == 0.0:
+        return None
+    return value
 
-def remove_ppa_division (division, row):
-    if row['issuing_agency'] == 'PPA':
-        return ''
+def parse_datetime(value):
+    if value[-5:] == '24:00':
+        value = value[:-5] + '23:59'
+    return datetime.strptime(value, '%m/%d/%Y%H:%M')
+
+typemap = {
+    'integer': int,
+    'string': lambda x: x.strip(),
+    'decimal': lambda x: Decimal(x.replace(',','')),
+    'numeric': float,
+    'datetime': parse_datetime,
+    'latlon': parse_lat_lon
+}
+
+def get_transform_row(latlon_input):
+    def transform_row(line):
+        row = OrderedDict()
+        for field in fieldmap.keys():
+            config = fieldmap[field]
+            if config['type'] == 'latlon' and not latlon_input:
+                row[field] = None
+            elif 'start_pos' in config and 'end_pos' in config:
+                row[field] = typemap[config['type']]\
+                                (line[0][config['start_pos']:config['end_pos']])
+            else:
+                row[field] = None
+
+        if not latlon_input:
+            row['lat'] = None
+            row['lon'] = None
+
+        if (row['division'] == '0000' or row['division'] == '00'):
+            row['division'] = None
+
+        if row['issuing_agency'] == 'RED LI':
+            row['issuing_agency'] = 'PRISON'
+
+        if row['issuing_agency'] != 'PPA':
+            row['division'] = None
+
+        return row.values()
+
+    return transform_row
+
+def anonymize(row):
+    global plates_counter, ticket_numbers_counter
+
+    out_row = list(row)
+
+    plate_key = row[2] + row[3]
+    if plate_key in plates:
+        out_row[3] = plates[plate_key]['id']
+    else:
+        plates_counter += 1
+        plates[plate_key] = {
+            'id': plates_counter,
+            'plate': row[2],
+            'state': row[3],
+            'date_added': datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+        }
+        out_row[3] = plates_counter
+
+    ticket_number = row[0]
+    if ticket_number in ticket_numbers:
+        if dedup is True:
+            ticket_numbers_counter += 1
+            ticket_numbers[ticket_number + 'E'] = ticket_numbers_counter
+            logger.info('Duplicate detected on ticket number {} - Adding {}E'.format(
+                ticket_number, ticket_number))
+        else:
+            out_row[0] = ticket_numbers[ticket_number]
+    else:
+        ticket_numbers_counter += 1
+        ticket_numbers[ticket_number] = ticket_numbers_counter
+        out_row[0] = ticket_numbers_counter
+
+    return out_row
+
+def geocode(row):
+    global geocode_stats
+
+    out_row = list(row)
+
+    address_components = passyunk_parser.parse(row[5])
```

Comment:
Does this handle all of the other cleanup involved in the previous [python script](https://gist.github.com/anonymous/8aa3a643d6d78d09a1520de6777587d1), including converting lots like `TORRESDALE LOT` to `9645 JAMES ST`?